src/pages/index.js

```diff
@@ -70,31 +128,104 @@ function createCard(data) {
   const cardElement = new Card(
     { data },
     selectors.cardTemplate,
-    handleImageClick
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    handleDislikeClick
   );
   return cardElement.getView();
 }
 
 /*------- Event Handlers --------*/
 
-function handleImageClick(imgData) {
-  imageModal.open(imgData);
+function handleImageClick(title, image) {
+  imageModal.open(title, image);
 }
 
-function handleProfileFormSubmit(formValues) {
-  updateUserInfo(formValues);
-
+function handleProfileFormSubmit(inputs) {
+  const newName = inputs.name;
+  const newDescription = inputs.description;
+  api
+    .updateUserInfo(newName, newDescription)
+    .then((response) => {
+      console.log("Profile updated successfully:", response);
+      currentUserInfo.setUserInfo({
+        name: newName,
+        description: newDescription,
+      });
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+    });
   profileEdit.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well